src/hotspot/share/code/aotCodeCache.cpp

```diff
@@ -89,8 +89,14 @@ uint AOTCodeCache::max_aot_code_size() {
 }
 
 void AOTCodeCache::initialize() {
+#if !(defined(AMD64) || defined(AARCH64))
```

Comment:
Sounds like we need to also take care about Zero? E.g. `defined(ZERO) || (!defined(AMD64) && !defined(AARCH64))`?

Reply:
Or invert it and put it at `else`, reads better and matches other blocks around Hotspot.

```
#if !defined(ZERO) && (defined(AMD64) || defined(AARCH64))
...
#else
  log_info(aot, codecache, init)("AOT Code Cache is not supported on this platform.");
  AOTAdapterCaching = false;
  return;
#endif
```

Reply:
Right. Do we have check for Zero in tests?

Reply:
> Do we have check for Zero in tests?

I don't think we have  a good way to do this with `@requires`, so it is fine to skip it for this PR.

Reply:
I will keep current `#if/#else` order.